core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -285,24 +266,58 @@ private void resetAdapter() {
     searchPresenter.getRecentSearches();
   }
 
-  private ArrayAdapter<String> getDefaultAdapter() {
-    return new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1) {
-      @NonNull
-      @Override
-      public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
-        View row;
+  private DefaultAdapter getDefaultAdapter() {
+    return new DefaultAdapter();
+  }
 
-        if (convertView == null) {
-          row = LayoutInflater.from(parent.getContext())
-            .inflate(android.R.layout.simple_list_item_1, null);
-        } else {
-          row = convertView;
-        }
+  public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+    private Context context;
+    private AdapterView.OnItemClickListener onItemClickListener;
+    List<String> searchList;
+
+    @NonNull @Override
+    public DefaultAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+      View view = (View) LayoutInflater.from(parent.getContext())
+        .inflate(android.R.layout.simple_list_item_1, null);
```

Comment:
Do not pass null to an inflate call, this can break theming as the newly inflated view should inherit attributes from its parent read [this](https://www.bignerdranch.com/blog/understanding-androids-layoutinflater-inflate/)
tl:dr in recycler do `inflate(R.layout.thing, parent, false)`